api_tests/validation/etcd_test.go

```diff
@@ -44,105 +67,69 @@ var _ = Describe("Etcd validation tests", func() {
-				"Type":  Equal(field.ErrorTypeInvalid),
-				"Field": Equal("spec.backup.store.prefix"),
-			})))),
-			Entry("should forbid name equal to etcd's name", newEtcd("etcd-name"), ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+			Entry("should forbid invalid spec.backup.store", &v1alpha1.StoreSpec{
+				Prefix: "invalid",
+			}, ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
 				"Type":  Equal(field.ErrorTypeInvalid),
 				"Field": Equal("spec.backup.store.prefix"),
 			})))),
-			Entry("should validate succesfully the object", newEtcd(""), BeNil()),
+			Entry("should allow valid spec.backup.store", &v1alpha1.StoreSpec{
+				Prefix: fmt.Sprintf("%s--%s/%s", namespace, uuid, name),
+			}, BeNil()),
+			Entry("should allow nil spec.backup.store", nil, BeNil()),
 		)
-
 	})
 
 	Describe("#ValidateEtcdUpdate", func() {
-		It("should prevent updating anything if deletion time stamp is set", func() {
-			old := newEtcd("")
-
+		It("should prevent updating anything if deletion timestamp is set", func() {
 			now := metav1.Now()
-			old.DeletionTimestamp = &now
-			old.ResourceVersion = "1"
+			etcd.DeletionTimestamp = &now
+			etcd.ResourceVersion = "1"
 
-			newetcd := newUpdatableEtcd(old)
-			newetcd.DeletionTimestamp = &now
-			newetcd.Spec.Backup.Port = pointer.Int32Ptr(42)
+			newEtcd := etcd.DeepCopy()
+			newEtcd.ResourceVersion = "2"
+			newEtcd.Spec.Backup.Port = pointer.Int32Ptr(42)
 
-			errList := validation.ValidateEtcdUpdate(newetcd, old)
+			errList := validation.ValidateEtcdUpdate(newEtcd, etcd)
 
 			Expect(errList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
 				"Type":  Equal(field.ErrorTypeInvalid),
 				"Field": Equal("spec"),
 			}))))
 		})
 
-		It("Should prevent updating of spec.backup.store.prefix", func() {
-			old := newEtcd("")
-			old.ResourceVersion = "1"
+		It("should prevent updating spec.backup.store.prefix", func() {
+			etcd.ResourceVersion = "1"
 
-			newetcd := newUpdatableEtcd(old)
-			newetcd.Spec.Backup.Store.Prefix = newetcd.Spec.Backup.Store.Prefix + "--"
-			newetcd.ResourceVersion = "2"
+			newEtcd := etcd.DeepCopy()
+			newEtcd.ResourceVersion = "2"
+			newEtcd.Spec.Backup.Store.Prefix = namespace + "/" + name
 
-			errList := validation.ValidateEtcdUpdate(newetcd, old)
+			errList := validation.ValidateEtcdUpdate(newEtcd, etcd)
 
 			Expect(errList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
 				"Type":  Equal(field.ErrorTypeInvalid),
 				"Field": Equal("spec.backup.store.prefix"),
 			}))))
 		})
+	})
 
-		It("should allow updating everything else", func() {
-			old := newEtcd("")
-			old.ResourceVersion = "1"
+	It("should allow updating everything else", func() {
+		etcd.ResourceVersion = "1"
 
-			newetcd := newUpdatableEtcd(old)
-			newetcd.Status = v1alpha1.EtcdStatus{
-				Replicas: int32(42),
-			}
-			newetcd.Spec.Etcd = v1alpha1.EtcdConfig{}
-			newetcd.Spec.Common = v1alpha1.SharedConfig{}
-			newetcd.Spec.Replicas = 42
+		new := etcd.DeepCopy()
```

Comment:
Let's call this `newEtcd` for consistency with the rest.
```suggestion
		newEtcd := etcd.DeepCopy()
```